app/controllers/application_controller.rb

```diff
@@ -2,4 +2,14 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+
+protected
+
+  def authenticate_user!
```

Comment:
Devise has default implementation, that will redirect to new_user_session_path. So seems like this method is odd.